subsys/caf/modules/ble_adv.c

```diff
@@ -410,18 +410,15 @@ static void init(void)
 	if (IS_ENABLED(CONFIG_CAF_BLE_ADV_GRACE_PERIOD)) {
 		k_work_init_delayable(&grace_period_end, grace_period_fn);
 	}
-
-	/* We should not start advertising before ble_bond is ready.
-	 * Stay in disabled state. */
-
-	module_set_state(MODULE_STATE_READY);
 }
 
 static void start(void)
 {
 	int err = ble_adv_start(true);
 
-	if (err) {
+	if (!err) {
+		module_set_state(MODULE_STATE_READY);
+	} else {
 		module_set_state(MODULE_STATE_ERROR);
 	}
```

Comment:
If it is possible, please limit the number of negations:

```suggestion
	if (err) {
		module_set_state(MODULE_STATE_ERROR);
	} else {
		module_set_state(MODULE_STATE_READY);
	}
```

Reply:
I initially preferred to keep the happy-path at the top, but changed according to your suggestions